File: Delete Columns to Make Sorted/Delete_Columns_to_Make_Sorted.py
# Runtime: 236 ms, faster than 13.62% of Python3 online submissions for Delete Columns to Make Sorted.
# Memory Usage: 14.5 MB, less than 5.02% of Python3 online submissions for Delete Columns to Make Sorted.

class Solution:
    def minDeletionSize(self, A: List[str]) -> int:
        
        if not A:
            return 0
        
        res = 0
        
        for j in range(len(A[0])):
            for i in range(1, len(A)):
                if ord(A[i][j]) < ord(A[i - 1][j]):
                    res += 1
                    break
        
        return res

File: Delete Columns to Make Sorted/Delete_Columns_to_Make_Sorted.cpp
// Runtime: 60 ms, faster than 68.94% of C++ online submissions for Delete Columns to Make Sorted.
// Memory Usage: 13.1 MB, less than 17.60% of C++ online submissions for Delete Columns to Make Sorted.

class Solution 
{
public:
    int minDeletionSize(vector<string>& A) 
    {
        // 边界条件处理
        if (A.size() == 0) return 0;

        int res = 0;
        
        for (int j = 0; j < A[0].size(); j++)
        {
            for (int i = 1; i < A.size(); i++)
            {
                if (A[i][j] < A[i - 1][j])
                {
                    res++;
                    break;
                }
            }    
        }
        return res;
    }
};